pylossless/pipeline.py

```diff
@@ -1173,51 +1216,77 @@ def run_with_raw(self, raw):
+            # 5. Filtering
+            self.filter(message="Filtering")
 
-        # 4.flag epochs based on large Channel Stdev. across time
-        self.flag_noisy_epochs(message="Flagging Noisy Time periods")
+            if picks == "eeg":
+                # These steps are relevant only for EEG. For example,
+                # MEG channels don't get bridged or high impedance.
+                # Further, MEG doesn't have a montage so
+                # flag_uncorrelated_channels would crash. We could use
+                # mne.channels.read_layout() for MEG channel if we need
+                # at some point to implement such a functionality. But it
+                # seems irrelevant.
 
-        # 5. Filtering
-        self.filter(message="Filtering")
+                # 6. calculate nearest neighbort r values
+                msg = "Flagging uncorrelated channels"
+                data_r_ch = self.flag_uncorrelated_channels(message=msg, picks=picks)
 
-        # 6. calculate nearest neighbort r values
-        msg = "Flagging uncorrelated channels"
-        data_r_ch = self.flag_uncorrelated_channels(message=msg)
+                # 7. Identify bridged channels
+                msg = "Flagging Bridged channels"
+                self.flag_bridged_channels(data_r_ch, message=msg)
 
-        # 7. Identify bridged channels
-        self.flag_bridged_channels(data_r_ch, message="Flagging Bridged channels")
+            # 8. Flag rank channels
+            self.flag_rank_channel(data_r_ch, message="Flagging the rank channel")
 
-        # 8. Flag rank channels
-        self.flag_rank_channel(data_r_ch, message="Flagging the rank channel")
+            if picks == "eeg":
+                # 9. Calculate nearest neighbour R values for epochs
+                msg = "Flagging Uncorrelated epochs"
+                self.flag_uncorrelated_epochs(message=msg, picks=picks)
 
-        # 9. Calculate nearest neighbour R values for epochs
-        self.flag_uncorrelated_epochs(message="Flagging Uncorrelated epochs")
+            if self.config["ica"] is None:
+                # Skip ICA steps.
+                continue
 
-        # 10. Run ICA
-        self.run_ica("run1", message="Running Initial ICA")
+            # 10. Run ICA
+            self.run_ica("run1", message="Running Initial ICA", picks=picks)
 
-        # 11. Calculate IC SD
-        self.flag_noisy_ics(message="Flagging time periods with noisy IC's.")
+            # 11. Calculate IC SD
+            msg = "Flagging time periods with noisy IC's."
+            self.flag_noisy_ics(message=msg, picks=picks)
 
-        # 12. TODO: integrate labels from IClabels to self.flags["ic"]
-        self.run_ica("run2", message="Running Final ICA and ICLabel.")
+            # 12. TODO: integrate labels from IClabels to self.flags["ic"]
+            msg = "Running Final ICA and ICLabel."
+            self.run_ica("run2", message=msg, picks=picks)
```

Comment:
I think we stuffed the ICLabel routine into run_ica. 

This is a note that ICLabel only works with EEG data. So we should check that it does not crash if you pass it, say, only MEG channels.. If it does crash then we probably should add an  `if picks == "eeg"` clause before IClabel?

Reply:
Good point about ICLabel making sense only for EEG. Added the conditional as suggested.